pkg/webhook/topic_validator.go

```diff
@@ -175,9 +167,12 @@ func (s *webhookServer) checkExistingKafkaTopicCRs(ctx context.Context,
 		}
 
 		referredNamespace := kafkaTopic.Spec.ClusterRef.Namespace
-		if (kafkaTopic.GetNamespace() == clusterNamespace && referredNamespace == "") || referredNamespace == clusterNamespace {
-			foundKafkaTopic = &kafkaTopicList.Items[i]
-			break
+		referredName := kafkaTopic.Spec.ClusterRef.Name
+		if referredName == topic.Spec.ClusterRef.Name {
```

Comment:
The conditions in these two `if` statement could be combined into a single expression:
```
if referredName == topic.Spec.ClusterRef.Name &&
  (kafkaTopic.GetNamespace() == clusterNamespace && referredNamespace == "" || 
  referredNamespace == clusterNamespace) {
```

Reply:
Yes it can combined but for me it looks better to separate